src/panels/MissionsPanel/TimeLine/TimeLine.tsx

```diff
@@ -0,0 +1,281 @@
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { ActionIcon, Group } from '@mantine/core';
+import {
+  axisLeft,
+  scaleUtc,
+  select,
+  zoom,
+  ZoomBehavior,
+  zoomIdentity,
+  zoomTransform
+} from 'd3';
+
+import { useSubscribeToTime } from '@/api/hooks';
+import { ZoomInIcon, ZoomOutIcon, ZoomOutMapIcon } from '@/icons/icons';
+import { DisplayType, IconSize } from '@/types/enums';
+import { Phase } from '@/types/mission-types';
+
+import { DisplayedPhase } from '../MissionContent';
+
+import { ActivityCircle } from './ActivityCircle';
+import { TimeLineConfig } from './config';
+import { MileStonePolygon } from './MilestonePolygon';
+import { Phases } from './Phases';
+import { TimeArrow } from './TimeArrow';
+import { TimeIndicator } from './TimeIndicator';
+
+import './TimeLine.css';
+
+interface TimeLineProps {
+  allPhasesNested: Phase[][];
+  displayedPhase: DisplayedPhase;
+  missionOverview: Phase;
+  setDisplayedPhase: (phase: DisplayedPhase) => void;
+}
+
+export function TimeLine({
+  allPhasesNested,
+  displayedPhase,
+  missionOverview,
+  setDisplayedPhase
+}: TimeLineProps) {
+  const [scale, setScale] = useState(1);
+  const [translation, setTranslation] = useState(0);
+  const now = useSubscribeToTime();
+
+  const yAxisRef = useRef<SVGGElement | null>(null);
+  const svgRef = useRef<SVGSVGElement | null>(null);
+  const zoomRef = useRef<ZoomBehavior<SVGSVGElement, unknown> | null>(null);
+  const timeIndicatorRef = useRef<SVGRectElement | null>(null);
+
+  const {
+    minLevelWidth,
+    margin,
+    maxWidth,
+    maxScale,
+    height,
+    paddingGraph,
+    transitionDuration
+  } = TimeLineConfig;
+
+  // Depth of nesting for phases
+  const nestedLevels = allPhasesNested.length;
+
+  // Ensure graph is large enough to show all phases
+  const minWidth = minLevelWidth * nestedLevels + margin.left + margin.right;
+
+  // Height of graph
+  // TODO anden88: right now we don't have a way to get the size of the panel we're in so we can't
+  // do any updates related to a panel resize and therefore use a fixed height.
+  // When height changes of window, rescale y axis
+  // const height = fullHeight - zoomButtonHeight;
```

Comment:
Now that we can get the height, do we want to do this straight away or in the future?